packages/release-mastermind/src/contexts/pullRequests.ts

```diff
@@ -9,139 +1,292 @@
-          'APPROVE'
-        )
-        return true
-      } else {
-        core.setFailed(convention.failedComment)
-        return false
-      }
-    })
-  }
-
-  bumpVersion(labels: Release['labels']) {
-    if (!labels || !this.context.props.labels) return
-    if (
-      (this.configs.versioning.type == 'SemVer' ||
-        this.configs.versioning == undefined) &&
-      this.newVersion.semantic
-    ) {
-      if (
-        this.context.props.labels[labels.major] || labels.breaking
-          ? this.context.props.labels[labels.major]
-          : true
-      ) {
-        this.newVersion.semantic.major++
-      } else if (this.context.props.labels[labels.minor]) {
-        this.newVersion.semantic.minor++
-      } else if (this.context.props.labels[labels.patch]) {
-        this.newVersion.semantic.patch++
-      }
-      if (this.context.props.labels[labels.prerelease]) {
-        this.newVersion.semantic.prerelease =
-          this.newVersion.semantic.prerelease ||
-          this.configs.prereleaseName ||
-          'prerelease'
-      }
-      if (this.context.props.labels[labels.build]) {
-        this.newVersion.semantic.build = +1
-      }
-      this.newVersion.name = `${this.newVersion.semantic.major}.${
-        this.newVersion.semantic.minor
-      }.${this.newVersion.semantic.patch}${
-        this.newVersion.semantic.prerelease
-          ? `-${this.newVersion.semantic.prerelease}`
-          : ''
-      }${
-        this.newVersion.semantic.build
-          ? `+${this.newVersion.semantic.build}`
-          : ''
-      }`
-      log(LoggingLevels.debug, `New Version is: ${this.newVersion.name}`)
-    }
-  }
+	context: PRContext
+	config: PullRequestConfig
+	constructor(
+		util: Utils,
+		runners: Runners,
+		configs: Config,
+		curContext: CurContext,
+		dryRun: boolean
+	) {
+		if (curContext.type !== "pr")
+			throw new LoggingDataClass(
+				LoggingLevels.error,
+				"Cannot construct without pr context"
+			)
+		super(util, runners, configs, curContext, dryRun)
+		this.context = curContext.context
+		if (!configs.pr)
+			throw new LoggingDataClass(
+				LoggingLevels.error,
+				"Cannot start without config"
+			)
+		this.config = configs.pr
+	}
+
+	/**
+	 * Parse the PR Context
+	 * @author IvanFon, TGTGamer, jbinda
+	 * @since 1.0.0
+	 */
+	static async parse(
+		utils: Utils,
+		config: Config,
+		context: Context
+	): Promise<PRContext | undefined> {
+		const pr = context.payload.pull_request
+		if (!pr) {
+			return
+		}
+
+		log(
+			LoggingLevels.debug,
+			`context.payload.pull_request: ` +
+				JSON.stringify(context.payload.pull_request)
+		)
+
+		const IDNumber = pr.number
+		const labels = await utils.parsingData.labels(pr.labels).catch((err) => {
+			log(LoggingLevels.error, `Error thrown while parsing labels: `, err)
+			throw err
+		})
+		const files: string[] = await utils.api.files
+			.list(IDNumber)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while listing files: `, err)
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Videndum/action-masterminds/pullRequest?prid=7872005)
```suggestion
      log(LoggingLevels.error, "Error thrown while listing files: ", err)
```